src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -300,8 +340,25 @@ const UsersFragment = ({
         <Text style={styles.noDataFoundText}>No users found</Text>
       </View>
     );
-
-  return <View>{renderUsersItem()}</View>;
+  return (
+    <View>
+      {loadingDM && (
+        <View
+          style={{
+            position: 'absolute',
+            zIndex: 1,
+            width: '100%',
+            height: '100%',
+            justifyContent: 'center',
+            alignContent: 'center',
+            alignSelf: 'center'
+          }}>
+          <ActivityIndicator size="large" />
+        </View>
+      )}
+      {renderUsersItem()}
+    </View>
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The loading indicator implementation is correct but it can be simplified by using a ternary operator instead of an AND operator. This makes the code more readable and easier to understand.
```diff
- {loadingDM && (
-   <View
-     style={{
-       position: 'absolute',
-       zIndex: 1,
-       width: '100%',
-       height: '100%',
-       justifyContent: 'center',
-       alignContent: 'center',
-       alignSelf: 'center'
-     }}>
-     <ActivityIndicator size="large" />
-   </View>
- )}
+ {loadingDM ? (
+   <View
+     style={{
+       position: 'absolute',
+       zIndex: 1,
+       width: '100%',
+       height: '100%',
+       justifyContent: 'center',
+       alignContent: 'center',
+       alignSelf: 'center'
+     }}>
+     <ActivityIndicator size="large" />
+   </View>
+ ) : renderUsersItem()}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->